File: aws_fargate/fetchTeeTimes-CPS/Course.ts
export class Course {
  id: number;
  name: string;
  display_name: string;
  club_name: string;
  city: string;
  rating: number;
  external_api: string;
  external_api_attributes: Record<string, any>;
  booking_visibility_days: number;
  requires_login: boolean;
  booking_visibility_start_time: string;
  timezone: string;

  constructor(
    id: number,
    name: string,
    display_name: string,
    club_name: string,
    city: string,
    rating: number,
    external_api: string,
    external_api_attributes: Record<string, any> = {},
    booking_visibility_days: number = 0,
    requires_login: boolean = false,
    booking_visibility_start_time: string = "",
    timezone: string = ""
  ) {
    this.id = id;
    this.name = name;
    this.display_name = display_name;
    this.club_name = club_name;
    this.city = city;
    this.rating = rating;
    this.external_api = external_api;
    this.external_api_attributes = external_api_attributes;
    this.booking_visibility_days = booking_visibility_days;
    this.requires_login = requires_login;
    this.booking_visibility_start_time = booking_visibility_start_time;
    this.timezone = timezone;
  }

  // Helper method to create a Course from a plain object
  static fromObject(obj: any): Course {
    // Handle nested city data from join
    const cityName = obj.cities?.name;
    
    return new Course(
      obj.id,
      obj.name,
      obj.display_name,
      obj.club_name,
      cityName,
      obj.rating,
      obj.external_api,
      obj.external_api_attributes || {},
      obj.booking_visibility_days || 0,
      obj.requires_login || false,
      obj.booking_visibility_start_time || "",
      obj.timezone || ""
    );
  }

  // Helper method to convert Course to a plain object
  toObject(): any {
    return {
      id: this.id,
      name: this.name,
      display_name: this.display_name,
      club_name: this.club_name,
      city: this.city,
      rating: this.rating,
      external_api: this.external_api,
      external_api_attributes: this.external_api_attributes,
      booking_visibility_days: this.booking_visibility_days,
      requires_login: this.requires_login,
      booking_visibility_start_time: this.booking_visibility_start_time,
      timezone: this.timezone
    };
  }
} 

File: aws_fargate/chrono/course.py
"""Course data class for ChronoGolf integration"""
from typing import Dict, Any, Optional
from dataclasses import dataclass


@dataclass
class Course:
    """Represents a golf course with all its booking and API attributes"""
    
    id: int
    name: str
    display_name: str
    club_name: str
    city: str
    rating: float
    external_api: str
    external_api_attributes: Dict[str, Any]
    booking_visibility_days: int = 0
    requires_login: bool = False
    booking_visibility_start_time: str = ""
    timezone: str = ""

    @classmethod
    def from_dict(cls, data: Dict[str, Any]) -> 'Course':
        """Create a Course instance from a dictionary (e.g., Supabase response)
        
        Args:
            data: Dictionary containing course data from database
            
        Returns:
            Course instance
        """
        # Handle nested city data from join
        city_name = data.get('cities', {}).get('name', '') if isinstance(data.get('cities'), dict) else ''
        
        return cls(
            id=data['id'],
            name=data['name'],
            display_name=data['display_name'],
            club_name=data['club_name'],
            city=city_name,
            rating=data['rating'],
            external_api=data['external_api'],
            external_api_attributes=data.get('external_api_attributes', {}),
            booking_visibility_days=data.get('booking_visibility_days', 0),
            requires_login=data.get('requires_login', False),
            booking_visibility_start_time=data.get('booking_visibility_start_time', ''),
            timezone=data.get('timezone', '')
        )

    def to_dict(self) -> Dict[str, Any]:
        """Convert Course to a dictionary
        
        Returns:
            Dictionary representation of the Course
        """
        return {
            'id': self.id,
            'name': self.name,
            'display_name': self.display_name,
            'club_name': self.club_name,
            'city': self.city,
            'rating': self.rating,
            'external_api': self.external_api,
            'external_api_attributes': self.external_api_attributes,
            'booking_visibility_days': self.booking_visibility_days,
            'requires_login': self.requires_login,
            'booking_visibility_start_time': self.booking_visibility_start_time,
            'timezone': self.timezone
        }

